### PR TITLE
Improvement/arsnn 1 export default configuration for maxsockets

### PR DESCRIPTION
**Pull request by ghost**
The goal is to export the default configuration, to be compatible with a flexible usage of the library along with others like `agentkeepalive`, while enforcing the default configuration in the same way.

### CODE REVIEW

**Review by jonathan-gramain — COMMENTED** [review]
I believe we could as well use `agentkeepalive` as a basis to keep the philosophy of embedding the default config automatically. As far as I can remember, it should be a drop-in replacement for http/https and it has been used in S3C for a while without issue so it should be safe.

File: lib/config/maxSockets.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ */
+const maxSocketsNumber = Number(process.env.MAX_SOCKETS) || 50;
+
+/**
+ * The default configuration for the maxSocket profile.
+ * The goal is to enforce a `maxSockets` property to properly
+ * handle load.
+ */
+const maxSocketsConfiguration: http.AgentOptions  = {
+    keepAlive: true,
```
Comment (jonathan-gramain) [inline lib/config/maxSockets.ts]: It's a bit strange to see `keepAlive` in a config named `maxSocketsConfiguration`, could it be named like `agentConfiguration`?

Comment (jonathan-gramain) [reply lib/config/maxSockets.ts]: And maybe rename the file as well

Comment (ghost) [reply lib/config/maxSockets.ts]: Thank you, I did it in https://github.com/scality/httpagent/pull/6/commits/eb40166fd867aef06827b6e6efcc51419cf2e550, and I used the name you suggested. I also updated the name of the exported variable.

**Comment by ghost** [comment]
> I believe we could as well use `agentkeepalive` as a basis to keep the philosophy of embedding the default config automatically. As far as I can remember, it should be a drop-in replacement for http/https and it has been used in S3C for a while without issue so it should be safe.

You are right, what I did is to introduce agentkeepalive and update the classes to extends the agentkeepalive's one. The default parameters will take place, by default. If needed, we can always override them in the projects.

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**